third_party/cpgfunctionEP-0.3.2/CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+cmake_minimum_required(VERSION 3.6)
+project(cpgfunctionEP)
+
+set(CMAKE_CXX_STANDARD 17)
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=c++17")
+else()
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Comment:
https://cmake.org/cmake/help/v3.4/variable/CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS.html

`CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` is the default value for `WINDOWS_EXPORT_ALL_SYMBOLS` which itself is "only implemented for MS-compatible tools on Window".

Perhaps the block should just be this?

```
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
if (MSVC)
  add_compile_definitions(_USE_MATH_DEFINES)
endif()
```

Reply:
Believed to be addressed in a7f66c005851db3116f729a4ad814d4d86fd0d41